src/test/scala/chiselTests/TesterDriverSpec.scala

```diff
@@ -0,0 +1,40 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+/**
+  * Created by chick on 2/9/16.
+  */
+import Chisel._
+import Chisel.testers.BasicTester
+
+/**
+  * Created by chick on 2/8/16.
+  */
+class FinishTester extends BasicTester {
+  var finish_was_run = false
+
+  override def finish(): Unit = {
+    finish_was_run = true
```

Comment:
I'd like to do something like this but it is difficult given the overall test harness now contained in chisel3.  If you have some idea of how to show this, I can try to implement